pytorch_lightning/profiler/profilers.py

```diff
@@ -90,13 +108,25 @@ def describe(self) -> None:
         """Logs a profile report after the conclusion of the training run."""
         for write in self.write_streams:
             write(self.summary())
+        if self.output_file:
+            self.output_file.flush()
 
-    @abstractmethod
-    def summary(self) -> str:
-        """Create profiler summary in text format."""
-
-    def on_train_start(self, local_rank: Optional[int] = None):
-        self.local_rank = local_rank
+    def stats_to_str(self, stats: dict) -> str:
```

Comment:
what's the structure expected in `stats` ? is this more of a utility that should exist independent of the base class?

Reply:
stats should be a dict[str, str]. It adds the rank is available, so I think it is fine in the BaseClass as all reports should have the same structure.